### PR TITLE
Migrate circleci configuration to 2.0

### PR DESCRIPTION
**Pull request by kalys**

### CODE REVIEW

**Comment by joshchernoff** [comment]
Why was there a change made to mix.lock ?

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - save_cache:
+          key: v1-build-cache-{{ .Branch }}
+          paths: "_build"
+      - save_cache:
+          key: v1-build-cache
+          paths: "_build"
+      - run: mix test
+      - run: MIX_ENV=prod mix compile
+
+      - store_test_results:  # upload test results for display in Test Summary
```
Comment (joshchernoff) [inline .circleci/config.yml]: Not sure storing test results is necessary, though I could be wrong.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+          key: v1-mix-cache
+          paths: "deps"
+      - save_cache:
+          key: v1-build-cache-{{ .Branch }}
+          paths: "_build"
+      - save_cache:
+          key: v1-build-cache
+          paths: "_build"
+      - run: mix test
+      - run: MIX_ENV=prod mix compile
```
Comment (joshchernoff) [inline .circleci/config.yml]: Needed?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      - run: mix do deps.get, compile
+      - save_cache:
+          key: v1-mix-cache-{{ .Branch }}-{{ checksum "mix.lock" }}
+          paths: "deps"
+      - save_cache:
+          key: v1-mix-cache-{{ .Branch }}
+          paths: "deps"
+      - save_cache:
+          key: v1-mix-cache
+          paths: "deps"
```
Comment (paulcsmith) [inline .circleci/config.yml]: I haven't used Circle 2.0 that much yet. Could you explain why multiple caches are needed here?

File: .circleci/config.yml (hunk 2 above)
Comment (paulcsmith) [reply .circleci/config.yml]: I've found this is helpful to catch things that may only cause issues in prod.

**Comment by kalys** [comment]
@joshchernoff @paulcsmith 
I used this file https://circleci.com/docs/2.0/language-elixir/#sample-configuration

Maybe cost of network traffic is higher than cost of storage and that's why they recommend to cache every version, branch and change of mix.lock.

**Comment by kalys** [comment]
@joshchernoff @paulcsmith Ok. I removed all caches except one that depends on checksum of mix.lock. Also I removed ```v1-``` prefix from cache name.

**Comment by paulcsmith** [comment]
Thank you so much!

**Pull request merged**